server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -465,6 +480,11 @@ private void setLowLevelCancellation(Boolean lowLevelCancellation) {
         this.lowLevelCancellation = lowLevelCancellation;
     }
 
+    private void setMaxClauseCount(int maxClauseCount) {
+        this.maxClauseCount = maxClauseCount;
+        IndexSearcher.setMaxClauseCount(maxClauseCount);
```

Comment:
Ok, this is the only shaky place I see: trying to apply volatile semantics (`maxClauseCount`) to non-volatile  static variable (`IndexSearcher::maxClauseCount`). It is certainly safer to keep this setting static, @msfroh wdyt?

Reply:
@reta is there any way forward other than making `IndexSearcher::maxClauseCount` volatile within lucene itself? I understand the semantics of this affect per shard state where some shards will reference the old clause count and new ones will have the new count but eventually consistency will prevail

Reply:
> @reta is there any way forward other than making `IndexSearcher::maxClauseCount` volatile within lucene itself?

No (AFAIK), we need changes in Apache Lucene (either non static args or thread friendly primitives). We just need to accept the limitation may be and document that behaviour.

Reply:
Yeah -- I'm not seeing what the local volatile `maxClauseCount` variable is doing for us in this case.

If the setting functionally just updates the Lucene static variable, that should be all that it does. (Also, the Lucene variable is not volatile, but I think it's fine for this setting to "eventually" take effect.)

Reply:
@harshavamsi let's just document that, please create a documentation issue for this change

Reply:
got it, I've reverted the volatile `max_clause_count` change, we just use the `IndeSearcher.setMaxClauseCount` now.